app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -1368,6 +1408,10 @@ public function saveProductEntity(array $entityRowsIn, array $entityRowsUp)
         }
         if ($entityRowsUp) {
             $this->_connection->insertOnDuplicate($entityTable, $entityRowsUp, ['updated_at', 'attribute_set_id']);
+            // Collecting existing product ids in variable to remove old ctegories
+            foreach ($entityRowsUp as $entityRowUp) {
+                $this->existingProductIds[] = $entityRowUp['entity_id'];
```

Comment:
this doesn't work on Magento Commerce.

you need to use `$linkFieldName = $this->getProductEntityLinkField();` which returns for Community `entity_id` and for Commerce `row_id`

Check `Import\Product` class for `$this->getProductEntityLinkField()` usages and review also other places where you added `entity_id`

Reply:
as table `catalog_category_product` has `product_id` column which is `entity_id` of product then `_saveProducts` needs extended to have entity_id always, i.e.

```diff
--- Model/Import/Product.php.orig	2019-11-27 08:08:03.903098754 +0100
+++ Model/Import/Product.php	2019-11-27 08:10:24.921517975 +0100
@@ -1638,11 +1638,18 @@
                         $attributeSetId = $this->skuProcessor->getNewSku($rowSku)['attr_set_id'];
                     }

-                    $entityRowsUp[] = [
+                    $existingRowItem = $this->getExistingSku($rowSku);
+                    $rowUp = [
                         'updated_at' => (new \DateTime())->format(DateTime::DATETIME_PHP_FORMAT),
                         'attribute_set_id' => $attributeSetId,
-                        $entityLinkField => $this->getExistingSku($rowSku)[$entityLinkField]
+                        $entityLinkField => $existingRowItem[$entityLinkField]
                     ];
+
+                    if ($entityLinkField !== 'entity_id') {
+                        $rowUp['entity_id'] = $existingRowItem['entity_id'];
+                    }
+
+                    $entityRowsUp[] = $rowUp;
                 } else {
                     if (!$productLimit || $productsQty < $productLimit) {
                         $entityRowsIn[strtolower($rowSku)] =
```

then this collecting to `existingProductIds` could stay same.